backend/FwLite/FwLiteProjectSync.Tests/Fixtures/SyncFixture.cs

```diff
@@ -83,4 +84,27 @@ public void DeleteSyncSnapshot()
         var snapshotPath = CrdtFwdataProjectSyncService.SnapshotPath(FwDataApi.Project);
         if (File.Exists(snapshotPath)) File.Delete(snapshotPath);
     }
+
+    private static readonly SemaphoreSlim _vernacularSemaphore = new(1, 1);
+
+    // a vernacular writing system is required in order to query for entries
+    // this is optional setup, because our core sync integration tests benefit from having a CRDT project that's as empty as possible
+    public async Task EnsureDefaultVernacularWritingSystemExistsInCrdt()
+    {
+        // This is optionally called from tests that consume this fixture, so it could get called multiple times in parallel
+        await _vernacularSemaphore.WaitAsync(100);
```

Comment:
I'm not sure this is doing what you expect. If there's already a writing system and no one has entered the semaphore, then it will enter immediately and then return without ever releasing it. The way it's written now all the code should be inside the try. If you don't start the try then the finally will never be called.